src/main/scala/introprog/IO.scala

```diff
@@ -51,7 +51,7 @@ object IO:
     * @param enc the encoding of the file.
     * */
   def saveLines(lines: Seq[String], fileName: String, enc: String = "UTF-8"): Unit =
-    saveString(lines.mkString("\n"), fileName, enc)
+    saveString(lines.mkString("\n") + "\n", fileName, enc)
```

Comment:
I think if lines are empty it should not write so better use `mkString("", "\n", "\n")` see docs for mkString to double-check the behavior or test in repl.